sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1625,6 +1625,17 @@ object SQLConf {
     .intConf
     .createWithDefault(25)
 
+  val MAX_PLAN_STRING_LENGTH = buildConf("spark.sql.maxPlanStringLength")
+    .doc("Maximum number of characters to output for a plan string.  If the plan is " +
+      "longer, further output will be truncated.  The default setting always generates a full " +
+      "plan.  Set this to a lower value such as 8192 if plan strings are taking up too much " +
+      "memory or are causing OutOfMemory errors in the driver or UI processes.")
+    .intConf
```

Comment:
Argh, should have commented before. But I think it's more user-friendly to use `bytesConf` here so that users can set the limit to e.g. "32k".

Also if the limit is `0` the plan string will be a little odd. Not a big deal though, but it could be improved (e.g. using a different message when it's 0).

Reply:
Fixed and changed the 0 length message to be more clear.